transaction/std/src/transaction_builder.rs

```diff
@@ -3104,4 +3105,505 @@ pub mod transaction_builder_tests {
             assert!(test_fn(block_version, tx_out1_wrong_amount).is_err());
         }
     }
+
+    #[test]
+    // Transaction builder with gift codes
+    fn test_gift_code_transactions() {
+        let mut rng: StdRng = SeedableRng::from_seed([1u8; 32]);
+        let block_version = BlockVersion::MAX;
+        let token_id = TokenId::from(5);
+        let fog_resolver = MockFogResolver::default();
+        let sender = AccountKey::random(&mut rng);
+        let receiver = AccountKey::random(&mut rng);
+        let (funding_input_amt, funding_output_amt, fee) = (1000, 10, 1);
+        let sender_reserved_destinations = ReservedDestination::from(&sender);
+        let receiver_reserved_destinations = ReservedDestination::from(&receiver);
```

Comment:
Any reason this is plural and not singular `destination`?

Reply:
Because the struct holds methods for multiple destinations?

Reply:
The struct itself should probably be renamed ReservedAddresses